### PR TITLE
[android] Tag JNI methods and enable R8 optimizations

### PR DESCRIPTION
**Pull request by rtsisyk**
R8 is not enabled for Debug, please test Beta or Release:

```
./gradlew -Parm64 runFdroidBeta
```

Results:

```
48534296 fdroid-beta-before.apk
45663208 fdroid-beta-after.apk
```

This patch saves almost 3MB or ~6%.

### CODE REVIEW

**Review by biodranik — APPROVED** [review]
It should be carefully tested. Let's release it in a beta together with KMB fix (in the master) and street TTS announcements (on the branch).

File: android/app/src/main/java/app/organicmaps/car/CarAppSession.java [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import android.content.Intent;
 import android.content.res.Configuration;
 
+import androidx.annotation.Keep;
```
Comment (strump) [inline android/app/src/main/java/app/organicmaps/car/CarAppSession.java:8]: Unused import?

File: android/app/src/main/java/app/organicmaps/downloader/CountrySuggestFragment.java [hunk 2]
```diff
@@ -10,6 +10,7 @@
 import android.widget.LinearLayout;
 import android.widget.TextView;
 
+import androidx.annotation.Keep;
```
Comment (strump) [inline android/app/src/main/java/app/organicmaps/downloader/CountrySuggestFragment.java:13]: Unused import?

File: android/app/src/main/java/app/organicmaps/downloader/DownloaderAdapter.java [hunk 3]
```diff
@@ -13,6 +13,7 @@
 import android.view.ViewGroup;
 import android.widget.TextView;
 
+import androidx.annotation.Keep;
```
Comment (strump) [inline android/app/src/main/java/app/organicmaps/downloader/DownloaderAdapter.java:16]: Unused import?

File: android/app/src/main/java/app/organicmaps/downloader/OnmapDownloader.java [hunk 4]
```diff
@@ -7,6 +7,7 @@
 import android.widget.Button;
 import android.widget.TextView;
 
+import androidx.annotation.Keep;
```
Comment (strump) [inline android/app/src/main/java/app/organicmaps/downloader/OnmapDownloader.java:10]: Unused import?

File: android/app/src/main/java/app/organicmaps/routing/RoutingMapsDownloadFragment.java [hunk 5]
```diff
@@ -6,6 +6,7 @@
 import android.view.View;
 import android.view.ViewGroup;
 
+import androidx.annotation.Keep;
```
Comment (strump) [inline android/app/src/main/java/app/organicmaps/routing/RoutingMapsDownloadFragment.java:9]: Unused import?

File: android/app/src/main/java/app/organicmaps/widget/placepage/PlacePageView.java [hunk 6]
```diff
@@ -15,6 +15,7 @@
 import android.widget.TextView;
 
 import androidx.annotation.IdRes;
+import androidx.annotation.Keep;
```
Comment (strump) [inline android/app/src/main/java/app/organicmaps/widget/placepage/PlacePageView.java:18]: Unused import?

**Review by strump — COMMENTED** [review]
Some strange imports found

**Review by Jean-BaptisteC — CHANGES_REQUESTED** [review]
❌ Pixel 6 - Android 13

- Search location
- Select arrow and add to bookmarks
- Open place page
- Edit bookmarks
- Switch to another list and change color of list
- Go back to map and leave place page
- Open bookmarks bottom menu
- Open list contain bookmark ->crash

```
10-04 12:13:12.946 E/AndroidRuntime(14960): FATAL EXCEPTION: main
10-04 12:13:12.946 E/AndroidRuntime(14960): Process: app.organicmaps.beta, PID: 14960
10-04 12:13:12.946 E/AndroidRuntime(14960): java.lang.RuntimeException: Unable to start activity ComponentInfo{app.organicmaps.beta/app.organicmaps.bookmarks.BookmarkListActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Class.isInterface()' on a null object reference
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3644)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3781)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:138)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2306)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Handler.dispatchMessage(Handler.java:106)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Looper.loopOnce(Looper.java:201)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Looper.loop(Looper.java:288)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.ActivityThread.main(ActivityThread.java:7918)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at java.lang.reflect.Method.invoke(Native Method)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
10-04 12:13:12.946 E/AndroidRuntime(14960): Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Class.isInterface()' on a null object reference
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at java.lang.Class.isAssignableFrom(Class.java:824)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Parcel.readParcelableCreatorInternal(Parcel.java:4900)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Parcel.readParcelableInternal(Parcel.java:4807)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Parcel.readValue(Parcel.java:4567)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Parcel.readValue(Parcel.java:4347)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Parcel.-$$Nest$mreadValue(Unknown Source:0)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Parcel$LazyValue.apply(Parcel.java:4445)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Parcel$LazyValue.apply(Parcel.java:4404)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.BaseBundle.getValueAt(BaseBundle.java:394)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.BaseBundle.getValue(BaseBundle.java:374)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.BaseBundle.getValue(BaseBundle.java:357)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.BaseBundle.get(BaseBundle.java:696)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Bundle.getParcelable(Bundle.java:953)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at androidx.activity.n.l(SourceFile:1)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at app.organicmaps.util.Utils.getParcelable(SourceFile:19)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at h2.r.z0(SourceFile:21)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at h2.r.R(SourceFile:4)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at androidx.fragment.app.w0.e(SourceFile:71)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at androidx.fragment.app.w0.k(SourceFile:149)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at androidx.fragment.app.p0.z(SourceFile:136)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at androidx.fragment.app.p0.T(SourceFile:82)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at androidx.fragment.app.p0.x(SourceFile:74)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at g2.c.V(SourceFile:57)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at g2.c.T(SourceFile:29)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at g2.f.T(SourceFile:1)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at g2.c.onCreate(SourceFile:150)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.Activity.performCreate(Activity.java:8342)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.Activity.performCreate(Activity.java:8321)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1417)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3625)
10-04 12:13:12.946 E/AndroidRuntime(14960): 	... 12 more
10-04 12:13:39.105 E/
```

File: android/app/src/main/java/app/organicmaps/car/CarAppSession.java (hunk 1 above)
Comment (rtsisyk) [reply android/app/src/main/java/app/organicmaps/car/CarAppSession.java]: Fixed.

**Comment by rtsisyk** [comment]
> ❌ Pixel 6 - Android 13
> 
>     * Search location
> 
>     * Select arrow and add to bookmarks
> 
>     * Open place page
> 
>     * Edit bookmarks
> 
>     * Switch to another list and change color of list
> 
>     * Go back to map and leave place page
> 
>     * Open bookmarks bottom menu
> 
>     * Open list contain bookmark ->crash
> 
> 
> ```
> 10-04 12:13:12.946 E/AndroidRuntime(14960): FATAL EXCEPTION: main
> 10-04 12:13:12.946 E/AndroidRuntime(14960): Process: app.organicmaps.beta, PID: 14960
> 10-04 12:13:12.946 E/AndroidRuntime(14960): java.lang.RuntimeException: Unable to start activity ComponentInfo{app.organicmaps.beta/app.organicmaps.bookmarks.BookmarkListActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Class.isInterface()' on a null object reference
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3644)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3781)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:138)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2306)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Handler.dispatchMessage(Handler.java:106)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Looper.loopOnce(Looper.java:201)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Looper.loop(Looper.java:288)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.ActivityThread.main(ActivityThread.java:7918)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at java.lang.reflect.Method.invoke(Native Method)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Class.isInterface()' on a null object reference
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at java.lang.Class.isAssignableFrom(Class.java:824)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Parcel.readParcelableCreatorInternal(Parcel.java:4900)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Parcel.readParcelableInternal(Parcel.java:4807)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Parcel.readValue(Parcel.java:4567)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Parcel.readValue(Parcel.java:4347)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Parcel.-$$Nest$mreadValue(Unknown Source:0)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Parcel$LazyValue.apply(Parcel.java:4445)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Parcel$LazyValue.apply(Parcel.java:4404)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.BaseBundle.getValueAt(BaseBundle.java:394)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.BaseBundle.getValue(BaseBundle.java:374)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.BaseBundle.getValue(BaseBundle.java:357)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.BaseBundle.get(BaseBundle.java:696)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.os.Bundle.getParcelable(Bundle.java:953)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at androidx.activity.n.l(SourceFile:1)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at app.organicmaps.util.Utils.getParcelable(SourceFile:19)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at h2.r.z0(SourceFile:21)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at h2.r.R(SourceFile:4)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at androidx.fragment.app.w0.e(SourceFile:71)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at androidx.fragment.app.w0.k(SourceFile:149)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at androidx.fragment.app.p0.z(SourceFile:136)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at androidx.fragment.app.p0.T(SourceFile:82)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at androidx.fragment.app.p0.x(SourceFile:74)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at g2.c.V(SourceFile:57)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at g2.c.T(SourceFile:29)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at g2.f.T(SourceFile:1)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at g2.c.onCreate(SourceFile:150)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.Activity.performCreate(Activity.java:8342)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.Activity.performCreate(Activity.java:8321)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1417)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3625)
> 10-04 12:13:12.946 E/AndroidRuntime(14960): 	... 12 more
> 10-04 12:13:39.105 E/
> ```

Fixed by switching to BundleCompat.getParcelableExtra().

**Comment by Jean-BaptisteC** [comment]
Pixel 6 - Android 14
- Start navigation
```
FATAL EXCEPTION: main
Process: app.organicmaps.beta, PID: 20467
java.lang.RuntimeException: Unable to start service app.organicmaps.routing.NavigationService@5a98a6f with Intent { cmp=app.organicmaps.beta/app.organicmaps.routing.NavigationService }: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{b45dff8 20467:app.organicmaps.beta/u0a615} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4839)
at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2289)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loopOnce(Looper.java:205)
at android.os.Looper.loop(Looper.java:294)
at android.app.ActivityThread.main(ActivityThread.java:8177)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
Caused by: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{b45dff8 20467:app.organicmaps.beta/u0a615} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
at android.os.Parcel.createException(Parcel.java:3041)
at android.os.Parcel.readException(Parcel.java:3024)
at android.os.Parcel.readException(Parcel.java:2966)
at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6761)
at android.app.Service.startForeground(Service.java:775)
at app.organicmaps.routing.NavigationService.onStartCommand(SourceFile:25)
at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4821)
at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0) 
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2289) 
at android.os.Handler.dispatchMessage(Handler.java:106) 
at android.os.Looper.loopOnce(Looper.java:205) 
at android.os.Looper.loop(Looper.java:294) 
at android.app.ActivityThread.main(ActivityThread.java:8177) 
at java.lang.reflect.Method.invoke(Native Method) 
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552) 
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971) 
Caused by: android.os.RemoteException: Remote stack trace:
at com.android.server.am.ActiveServices.validateForegroundServiceType(ActiveServices.java:2611)
at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:2322)
at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:1679)
at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:13265)
at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3385)
```

**Comment by rtsisyk** [comment]
> Pixel 6 - Android 14
> 
>     * Start navigation
> 
> 
> ```
> FATAL EXCEPTION: main
> Process: app.organicmaps.beta, PID: 20467
> java.lang.RuntimeException: Unable to start service app.organicmaps.routing.NavigationService@5a98a6f with Intent { cmp=app.organicmaps.beta/app.organicmaps.routing.NavigationService }: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{b45dff8 20467:app.organicmaps.beta/u0a615} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
> ```

This is a regression caused by SDK=34. I've removed that change from this PR and bumped `androidx.core` to 1.10.1 instead of 1.12.0. 1.10.1 already contains `BundleCompat.getParcelable()` which is needed to fix https://github.com/organicmaps/organicmaps/pull/6199#pullrequestreview-1657188721, but still doesn't require SDK=34. Thanks for your patience!

**Comment by rtsisyk** [comment]
let it be

**Pull request merged**